src/Microsoft.OData.Core/JsonLight/ODataJsonLightValueSerializer.cs

```diff
@@ -75,6 +97,87 @@ public virtual void WriteNullValue()
             }
         }
 
+        /// <summary>
+        /// Writes out the value of a resource (complex or entity).
+        /// </summary>
+        /// <param name="resourceValue">The resource (complex or entity) value to write.</param>
+        /// <param name="metadataTypeReference">The metadata type for the resource value.</param>
+        /// <param name="isTopLevel">true when writing a top-level property; false for nested properties.</param>
+        /// <param name="isOpenPropertyType">true if the type name belongs to an open (dynamic) property.</param>
+        /// <param name="duplicatePropertyNamesChecker">The checker instance for duplicate property names.</param>
+        /// <remarks>The current recursion depth should be a value, measured by the number of resource and collection values between
+        /// this resource value and the top-level payload, not including this one.</remarks>
+        [SuppressMessage("Microsoft.Naming", "CA2204:LiteralsShouldBeSpelledCorrectly", Justification = "Names are correct. String can't be localized after string freeze.")]
+        public virtual void WriteResourceValue(
+            ODataResourceValue resourceValue,
+            IEdmTypeReference metadataTypeReference,
+            bool isTopLevel,

```

Comment:
is an ODataResource ever top level?

Reply:
Same.  We don't limit end user to create a property with value as ODataResourceValue.
In my opinion, We should not limit end user to write a top level property with value as "ODataResourceValue".

Reply:
Now that we've prohibited writing resource values at the top level, can we get rid of this code?

---
In reply to: [241587501](https://github.com/OData/odata.net/pull/1346#discussion_r241587501) [](ancestors = 241587501)

Reply:
@mikepizzo removed. #Resolved.

Reply:
> In my opinion, We should not limit end user to write a top level property with value as "ODataResourceValue". 

Hi @xuzhg, we are heavily depending on creating "ODataResourceValue" at top level, but after upgrading to 7.6.4 we are unable to perform this operation. 

Our example URL format will look like below.

**URL**: root-url/Mailbox('4f89d8bc-4aaa-4aec-a459-5b5a7335b955')/Exchange.GetMailboxStatistics()

Here, the idea is Exchange.GetMailboxStatistics() operational segment should return MailboxStatistics, ODataResourceValue at top level. 

Our mail goal is not to change the URL format as lot of customers are rely on mentioned format.

Any suggested work around for this?  Please suggest me if I need to create an issue for this.

Thanks, 
Venu.

Reply:
@ve9u Why don't use binding (deserializer) to accept the top level complex ?

Reply:
Hi @xuzhg, Thanks for your help on this. writing the complex entity same as the writing entity set solves my issue. Later I found [this ](https://docs.microsoft.com/en-us/odata/odatalib/merge-complex-and-entity)blog post which explains the same thing.